src/clarity-angular/layout/nav/navLevelDirective.ts

```diff
@@ -3,18 +3,18 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {AfterContentInit, Directive, ElementRef, HostListener, Input} from "@angular/core";
+import {Directive, ElementRef, HostListener, Input, OnInit} from "@angular/core";
 
 import {ClrResponsiveNavCodes} from "./clrResponsiveNavCodes";
 import {ClrResponsiveNavigationService} from "./clrResponsiveNavigationService";
 
 @Directive({selector: "[clr-nav-level]"})
-export class NavLevelDirective implements AfterContentInit {
+export class NavLevelDirective implements OnInit {
     @Input("clr-nav-level") _level: number;
 
     constructor(private responsiveNavService: ClrResponsiveNavigationService, private elementRef: ElementRef) {}
 
-    ngAfterContentInit() {
+    ngOnInit() {
```

Comment:
Since we are moving this block of code execution from `ngAfterContentInit` to `ngOnInit`, do we need to use oompa-loompa as well? I am assuming that the chocolate error's root cause was due to the fact that we changed certain things after the view was initialized.

Reply:
@Shijir: The error is thrown even if we move it earlier in the lifecycle. I just moved it earlier because @youdz mentioned that user inputs are available in `ngOnInit` itself so I wanted to use them as early as possible.

Reply:
I see.

Reply:
@Shijir: Just realized that I didn't mention the cause of the issue before. The issue is happening because the header binding is being changed by the nav level directives after the header's view is already initialized.